controllers/datadoghq/tls.go

```diff
@@ -0,0 +1,232 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package datadoghq
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/DataDog/watermarkpodautoscaler/apis/datadoghq/v1alpha1"
+)
+
+const (
+	certificateCacheTimeout    = 10 * time.Minute
+	certificateCacheLRUTimeout = 10 * time.Minute
+)
+
+type tlsTransport struct {
+	wrappedTransport *http.Transport
+}
+
+// tlsCertificateCache is a LRU cache to store in memory certificates
+// used for TLS connections
```

Comment:
Do we really need LRU ? we could just have them expire ?

Reply:
It's not a real LRU as it's not memory bounded and entries are not expired until the certificate itself is used and we noticed that it expired. My comment is a lie :)
We can garbage collect the content and remove expired certificates, is that what you suggest?